src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -378,11 +385,21 @@ def update_existing_check_in(
     if processing_errors:
         raise ProcessingErrorsException(processing_errors, monitor=monitor)
 
-    updated_checkin: _CheckinUpdateKwargs = {
-        "status": updated_status,
-        "duration": updated_duration,
-        "date_updated": start_time,
-    }
+    updated_checkin: _CheckinUpdateKwargs = {}
+
+    if updated_duration:
+        updated_checkin["duration"] = updated_duration
```

Comment:
Is it ok to update the duration if there's one already set? I guess only if the user explicitly passed it?

Reply:
Yeah if they explicitly pass it we want to update.